tests/network-timeout.spec.ts

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env -S node --no-warnings --loader ts-node/esm
+
+import { sinon, test }  from 'tstest'
+
+import { FileBox } from '../src/mod.js'
+
+import { HTTP_TIMEOUT } from '../src/config.js'
+
+test('slow network stall HTTP_TIMEOUT', async t => {
+  const sandbox = sinon.createSandbox()
+  sandbox.useFakeTimers(Date.now())
+
+  const spy = sandbox.spy()
+
+  const stream = await FileBox
+    .fromUrl('https://www.google.com')
+    .toStream()
+
+  stream.on('error', spy)
+
+  const start = Date.now()
+
+  await sandbox.clock.tickAsync(HTTP_TIMEOUT - 1)
+  t.ok(spy.notCalled, `should not get error after TIMEOUT ${HTTP_TIMEOUT} - 1 (${Date.now() - start} passed)`)
+
+  await sandbox.clock.tickAsync(10)
+  t.ok(spy.calledOnce, `should get error after TIMEOUT ${HTTP_TIMEOUT} (${Date.now() - start} passed)`)
```

Comment:
Please help me to make this unit test work as expected if you have time, or wait for me to fix it later.

After passing this unit test, I believe this PR will be ready to be merged.

Thanks!

Reply:
I think the problem only occurs when there is a network failure, but I haven't thought of a way to simulate a network failure.

Reply:
I have updated the test and believe it works fine.